src/main/java/teammates/ui/template/AdminActivityLogTableRow.java

```diff
@@ -0,0 +1,304 @@
+        private String textClass;
+        
+        ActionTypeCssClass(String cClass, String tClass) {
+            buttonClass = cClass;
+            textClass = tClass;
+        }
+          
+        public String toHtmlButtonClass() {
+            return buttonClass;
+        }
+        
+        public String toHtmlTextClass() {
+            return textClass;
+        }
+        
+        public static ActionTypeCssClass generateActionTypeEnum(String action) {
+            switch(action) {
+            case Const.ACTION_RESULT_FAILURE:
+                return WARNING;
+            case Const.ACTION_RESULT_SYSTEM_ERROR_REPORT:
+                return DANGER;
+            default:
+                return NORMAL;
+            }
+        }
+    }
+    
+    private enum UserRoleCssClass {
+        ADMIN("glyphicon glyphicon-user text-danger"),
+        INSTRUCTOR("glyphicon glyphicon-user text-primary"),
+        STUDENT("glyphicon glyphicon-user text-warning"),
+        AUTO("glyphicon glyphicon-cog"),
+        UNREGISTERED("glyphicon glyphicon-user"),
+        UNKNOWN("");
+        
+        private String iconCssClass;
+        
+        UserRoleCssClass(String className) {
+            iconCssClass = className;
+        }
+        
+        public static UserRoleCssClass generateRoleCssClassHelper(String role) {
+            switch(role) {
+            case Const.ActivityLog.ROLE_ADMIN:
+                return ADMIN;
+            case Const.ActivityLog.ROLE_INSTRUCTOR:
+                return INSTRUCTOR;
+            case Const.ActivityLog.ROLE_STUDENT:
+                return STUDENT;
+            case Const.ActivityLog.ROLE_AUTO:
+                return AUTO;
+            default:
+                if (role.contains(Const.ActivityLog.ROLE_UNREGISTERED)) {
+                    return UNREGISTERED;
+                }
+                return UNKNOWN;
+            }
+        }
+        
+        public String getIconCssClass() {
+            return iconCssClass;
+        }
+    }
+    
+    private ActivityLogEntry activityLog;
+    
+    private String[] keyStringsToHighlight;
+    
+    public AdminActivityLogTableRow(ActivityLogEntry entry) {
+        activityLog = entry;
+    }
+    
+    public void setKeyStringsToHighlight(String[] strs) {
+        keyStringsToHighlight = strs;
+    }
+    
+    // --------------- Additional generated fields ---------------
+    
+    public String getUserHomeLink() {
+        switch(activityLog.getRoleWithoutMasquerade()) {
+        case Const.ActivityLog.ROLE_STUDENT:
+            return Url.addParamToUrl(Const.ActionURIs.STUDENT_HOME_PAGE,
+                    Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+        case Const.ActivityLog.ROLE_INSTRUCTOR:
+            return Url.addParamToUrl(Const.ActionURIs.INSTRUCTOR_HOME_PAGE,
+                    Const.ParamsNames.USER_ID, activityLog.getGoogleId());
+        default:
+            return activityLog.getGoogleId();
```

Comment:
A `null` return will make more sense here. There should be no "user home link" when there is no "user" in the first place, right?